tests/engines/collections/integration/components/collection-submission-confirmation-modal/component-test.ts

```diff
@@ -1,103 +1,183 @@
 import { click, render } from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
-import { t } from 'ember-intl/test-support';
+import { t, TestContext } from 'ember-intl/test-support';
 import { setupEngineRenderingTest } from 'ember-osf-web/tests/helpers/engines';
 import { module, test } from 'qunit';
 
 module('Integration | Component | collection-submission-confirmation-modal', hooks => {
     setupEngineRenderingTest(hooks, 'collections');
+    let externalSaveClicked = false;
+    let externalCancelClicked = false;
+    let externalResubmitClicked = false;
 
-    test('it renders', async function(assert) {
-        this.set('noop', () => { /* noop */ });
-        await render(hbs`
-        {{collection-submission-confirmation-modal
-            resubmitToCollection=(action this.noop)
-            openModal=true addToCollection=(action this.noop)
-            cancel=(action this.noop)
-        }}`);
-        assert.dom('[data-test-collection-submission-confirmation-modal-header]').hasText(
-            t('collections.collection_submission_confirmation_modal.header').toString(),
-        );
-        assert.dom('[data-test-collection-submission-confirmation-modal-body]').hasText(
-            t('collections.collection_submission_confirmation_modal.body').toString(),
+    hooks.beforeEach(async function(this: TestContext) {
+        externalSaveClicked = false;
+        externalCancelClicked = false;
+        externalResubmitClicked = false;
+
+        this.set('externalResubmitAction', () => {
+            externalResubmitClicked = true;
+        });
```

Comment:
An optional suggestion: You can do something here like `this.set('externalResubmitAction', sinon.fake())` to stub actions that you want to use in your tests. With `sinon.fake`, you can check to see if these handlers were called with the expected arguments by asserting something like `assert.ok(this.externalResubmitAction.calledWith('my comment')`, among other things. Sinon has a lot of cool things you can do in tests, so would recommend trying it out! https://sinonjs.org/releases/v15/fakes/